test/e2e/framework/test_context.go

```diff
@@ -221,6 +221,8 @@ type NodeTestContextType struct {
 	// the node e2e test. If empty, the default one (system.DefaultSpec) is
 	// used. The system specs are in test/e2e_node/system/specs/.
 	SystemSpecName string
+	// RestartKubelet restarts Kubelet unit when the process is killed.
+	RestartKubelet bool
```

Comment:
i don't think we should make the framework test context aware of such an option.
in fact a lot of options should be trimmed out of it if we eventually want to treat it as an API.

/cc @oomichi @alejandrox1

Reply:
@neolit123 As you stated this can be applied to other options in the `NodeTestContextType` struct, since it's part of TextContextType but only used on Node E2E scope. I don't see any immediate place to raise the variable, any suggestions?

Reply:
given a number of flags already use the framework context for storage, this one can go there as well.
https://github.com/kubernetes/kubernetes/blob/00da68dbc25d9d12b05200b9207abc14dde71246/test/e2e_node/e2e_node_suite_test.go#L81-L87

but long term this should be decoupled.
ideally the PR should pass review by SIG Node / e2e_node owners.

Reply:
@SergeyKanzhelev this is an open discussion, do reusing the TestContext for node tests to be reviewed, starting on this PR? 

I think @alejandrox1 started a document for this refactoring.

Reply:
I like the suggested refactoring, but I agree it is outside the scope of this PR. I think @neolit123 agreed with this. @ike-ma any concerns with making refactoring separately?

Reply:
> @SergeyKanzhelev this is an open discussion, do reusing the TestContext for node tests to be reviewed, starting on this PR?
> 
> I think @alejandrox1 started a document for this refactoring.

@alejandrox1 @knabben Do we have an issue for the refactoring?

Reply:
@SergeyKanzhelev @cynepco3hahue https://github.com/kubernetes/kubernetes/issues/101986